### PR TITLE
ML training script

### LINKED ISSUE #24: Create a ML script - functions - Tests

### PR DESCRIPTION
**Pull request by NikitaShymberg**
This pull request completes #24.

The code does not have full documentation yet or any tests. I will create an issue for that sometime later this week and will do that then. In the meantime I want to get this merged in so that everyone else can get going.

Please review my code!

### CODE REVIEW

**Comment by SonQBChau** [comment]
- I would image that in the report, we run the script and somehow be able to pull the table to ipynb to display it. So maybe a function to return dataframe? It can be very similar to your read_data function
- If I’m not mistaken, you are using best_estimator for r2 now since multiple metrics doesn’t work. My thinking is we can just show the table and manually pick the best params? Ex: Hey look, for this model we think MSE is the best metric, so we choose C = 1.0, gamma = 10.0
- Should we have 1 function run 6 models or 6 functions separately with modifiable params? So we can call whatever from the report.
- I know there is a requirement for saving the models, but do we need a function to load it later? We don’t need to do it, just thinking.
- Maybe a function to put every models in a dataframe and display them side by side, like results = {} in 573? Is it better to do from the report?

**Comment by NikitaShymberg** [comment]
Re 1: normally, I would agree with you. However, for this project we're supposed to format is such that each script is run sequentially (ultimately we're going to create a makefile that runs the scripts one by one). This is how I intended for this script to be used - run it from the command line and then read in the results.

Re 2: the problem with this approach is that it won't refit the model since the `RandomizedSearchCV` doesn't know which hyperparameters are best. I _can_ make this change if you think that it is best, but this means that you would need to fit the models manually in your script. I think that the proper way to do this is by setting the `refit` argument in `RandomizedSearchCV`, but for some reason I can't set it correctly for a regression model. Lastly, for this project, I think it's ok to just use R^2 score and move on.

Re 3: see my comment for number 1. This script is intended to be run independently, not just define functions to be used in other scripts.

Re 4: good idea! I've made this change.

Re 5: really I think this is more part of #22 but I've cleaned up the way that the results are saved to make this easier.


Let me know if there's anything else!

**Review by SonQBChau — APPROVED** [review]
